sdk/python/approzium/_postgres/__init__.py

```diff
@@ -0,0 +1,119 @@
+"""
+Internal module that implements Approzium routines that are common to Postgres
+libraries and the Postgres protocol.
+"""
+import logging
+import struct
+
+from ..authenticator import get_hash
+from ..misc import read_int32_from_bytes
+from .scram import SCRAMAuthentication
+
+logger = logging.getLogger(__name__)
+
+# Postgres protocol constants
+# derived from PGsource/src/include/libpq/pgcomm.h
+AUTH_REQ_MD5 = 5
+AUTH_REQ_SASL = 10
+
+
+def parse_msg(msg):
+    msg_type = msg[0]
```

Comment:
Would it be possible for `msg` to be `None`, causing an exception here? If so, is it normal in Python to leave these like this anyway, or is it normal to add a check? I can't remember 😊

Reply:
Since it is a positional argument, it's supposed to be required. Still, you can pass in `None`, but the common thinking is: at that point, you should know what you are doing and don't complain if you get an error. One reason for this is that `None` is its own type and entity really - unlike `nil` in Golang which AFAIK could be a value for multiple types. My impression about checking types or properties is that: people usually don't bother, because sooner or later an Exception will happen. For example, here, if msg is `None`, you would get a `TypeError`, which is not so bad. If we add a runtime check, we would probably raise the same exception type but put a different message. It won't be that big of a difference, and thus usually people don't bother. That is my current understanding but I am not an expert :)